app/Models/Multiplayer/UserScoreAggregate.php

```diff
@@ -133,6 +133,21 @@ public function updateUserAttempts()
         $this->increment('attempts');
     }
 
+    public function userRank()
+    {
+        if ($this->total_score === null || $this->last_score_id === null) {
+            return;
+        }
+
+        $query = $this->room->topScores()
```

Comment:
Annoyingly queries for the room again when room is already loaded in `RoomController`